src/web/handlers/status_list/update_token_status.rs

```diff
@@ -0,0 +1,187 @@
+use axum::{extract::State, response::IntoResponse, Extension, Json};
+use hyper::StatusCode;
+
+use crate::{
+    model::StatusListTokenPayload,
+    utils::{
+        bits_validation::BitFlag, errors::Error, lst_gen::update_status_list, state::AppState,
+    },
+};
+
+use super::error::StatusListError;
+
+// Handler to update an existing status list token
+pub async fn update_token_status(
+    State(appstate): State<AppState>,
+    Extension(issuer): Extension<String>,
+    Json(payload): Json<StatusListTokenPayload>,
+) -> Result<impl IntoResponse, StatusListError> {
+    let store = &appstate.status_list_token_repository;
+
+    let bits = BitFlag::new(payload.bits).ok_or_else(|| {
+        StatusListError::Generic(format!(
+            "Invalid 'bits' value: {}. Allowed values are 1, 2, 4, 8.",
+            payload.bits
+        ))
+    })?;
+
+    // Fetch the existing token
+    let token = match store.find_one_by(payload.list_id.clone()).await {
+        Ok(tokens) => tokens,
+        Err(e) => {
+            tracing::error!(error = ?e, list_id = ?payload.list_id, "Database query failed for status list.");
+            return Err(StatusListError::InternalServerError);
+        }
+    };
+    let token = match token {
+        Some(token) => token,
+        None => {
+            tracing::error!("Token not found in the database.");
+            return Err(StatusListError::StatusListNotFound);
+        }
+    };
+
+    // check if the request issuer matches the token issuer
+    if token.issuer != issuer {
+        tracing::error!("Issuer mismatch: expected {}, got {}", token.issuer, issuer);
+        return Err(StatusListError::IssuerMismatch);
+    }
+
+    // Update the status list
+    let updated_lst =
+        update_status_list(token.status_list.lst.clone(), payload.status.clone(), bits).map_err(
+            |e| {
+                tracing::error!("update_status_list failed: {:?}", e);
+                match e {
+                    Error::Generic(msg) => StatusListError::Generic(msg),
+                    Error::InvalidIndex => StatusListError::InvalidIndex,
+                    Error::UnsupportedBits => StatusListError::UnsupportedBits,
+                    _ => StatusListError::Generic(e.to_string()),
+                }
+            },
+        )?;
+
+    let mut exact_token = token;
+    exact_token.status_list.lst = updated_lst;
+    exact_token.status_list.bits = payload.bits;
+    exact_token.status_list.bits = payload.bits;
```

Comment:
This instruction appears twice.
```suggestion
    exact_token.status_list.bits = payload.bits;
```